traffic_ops/traffic_ops_golang/routing/routes.go

```diff
@@ -301,6 +301,8 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodGet, `regions/name/{name}/?(\.json)?$`, region.GetName, auth.PrivLevelReadOnly, Authenticated, nil, 503583197, noPerlBypass},
 		{1.1, http.MethodPut, `regions/{id}$`, api.UpdateHandler(&region.TORegion{}), auth.PrivLevelOperations, Authenticated, nil, 226308224, noPerlBypass},
 		{1.1, http.MethodPost, `regions/?$`, api.CreateHandler(&region.TORegion{}), auth.PrivLevelOperations, Authenticated, nil, 1288334488, noPerlBypass},
+		{1.5, http.MethodDelete, `regions/?$`, api.DeleteHandler(&region.TORegion{}), auth.PrivLevelOperations, Authenticated, nil, 2032626758, noPerlBypass},
+		{1.1, http.MethodDelete, `regions/name/{name}$`, handlerToFunc(proxyHandler), auth.PrivLevelReadOnly, Authenticated, nil, 1925881096, noPerlBypass},
```

Comment:
I think normally routes that use the `proxyHandler` have `NoAuth` because the assumption is that the Perl server will handle authentication.

Reply:
That makes sense, changed in c947d23d9.

Reply:
Why is the 1.1 route proxied to Perl? Reading the code, it appears the new handler can delete by either `id` or `name`, right?

Reply:
well this is just an explicit proxy to the deprecated handler in the perl, either way that route is deprecated and the notices were added in the perl implementation I didnt see any harm in having this in here

Reply:
Yeah I agree, seems like a fine way to ensure the deprecation notice. It saves him from needing to write a wrapper that inspects the request path.

Reply:
Ok, I guess usually the routes are just left out so be handled by the "route not found handler" which right now is "check TO-Perl". _Explicit_ `proxyHandler` routes are usually only there to handle requests explicitly before they're unintentionally handled by a different route (i.e. GET /cdns/foo being unintentionally handled by GET /cdns because the request matches it). That doesn't seem to be the case here, right?

Reply:
It's not, and the new handler can delete by either `id` or `name`. However, if we wanted to handle it in Go we would need to add a way for that handler to receive arbitrary alerts. The only shared handler I see that can already do that is `DeprecatedReadHandler()`, so doing that here would mean making a completely new DELETE handler just for this case.

Of course, it would be easier to just make a `region`-specific handler. In this case, though, the new handler cannot be used for a deprecated route.

Reply:
The reason I made it an explicit `proxyHandler` route was to keep `routes.go` useful as a reference for which routes exist. It would be super easy to remove, though.

Reply:
Ok, well `routes.go` might not even be a true list of all routes that exist currently because I'm not sure we've followed that pattern for other "deprecate but don't rewrite" routes. Not a big deal, just made me confused when I saw it.